component/azstorage/block_blob.go

```diff
@@ -980,6 +982,9 @@ func (bb *BlockBlob) stageAndCommitModifiedBlocks(name string, data []byte, offs
 }
 
 func (bb *BlockBlob) StageAndCommit(name string, bol *common.BlockOffsetList) error {
+	// lock on the blob name so that no stage and commit race condition occur causing failure
+	unlock := bb.blockLocks.Lock(name)
```

Comment:
Ambiguous in name "unlock := l....lock()". Rather we can just return the mutex from the method and let the caller decide whether to lock or unlock. Will be more readable code,

Reply:
done